draft-ietf-quic-transport.md

```diff
@@ -2116,6 +2116,9 @@ packets.
 A server conveys a preferred address by including the preferred_address
 transport parameter in the TLS handshake.
 
+Servers MAY communicate a preferred address of each address family (IPv4 and
```

Comment:
I think I would prefer to call this an "alternate" address.

Reply:
If you're speaking of "preferred" address, this is consistent with the rest of the text.  If you want to change the name, I'd suggest opening a separate issue.

I'm happy with preferred, since if it's just alternate, why would the client change?  Alternate implies no preference, when in fact the server does prefer the other address.

Reply:
"preferred" seems right.  Yes, it is an alternative, but the fact that it's a preferred alternative is what is important.

Reply:
I realize, looking back on the issue, why Jana mentioned alternate.  There are two separate cases: Connection migration and Server's Preferred Address.

In one case, you want to make sure connection migration doesn't fail when moving from v6 to v4 only, but you don't want them to do anything after the handshake.  In the other case, you want them to migrate to a more stable IP after the handshake.

If the server omits the address family you connected on, but supplies a preferred address on the other address family, do you try to migrate?

I suspect we need to change the structure from what's in this PR, and if not, we need to nail down the critical use cases and make sure they all work.

Reply:
Re-reading this PR, I think its workable as is, but this might read better if this paragraph was merged into the next paragraph and the next paragraph was updated to something like: "one of the servers preferred addresses using..."

Reply:
Yes, this ambiguity comes from the fact that this PR solves a problem that is slightly different than the one solved by the PreferredAddress TP.  I think you're all right that "preferred" is better than "alternate", but there's a gap.

I think this can be addressed by just a bit of text stating that it is up to the client to decide which among the preferred addresses it uses.  If the client has access to both address families, it's up to the client to decide whether it wants to migrate or not. The addresses are preferred by the server, but it's up to the client to decide if and when it moves to those.